netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -360,15 +358,24 @@ private int getAndIncrementNextStreamId() {
     return id;
   }
 
+  private Http2Stream requireHttp2Stream(int streamId) throws Http2Exception {
+    Http2Stream stream = connection().stream(streamId);
+    if (stream == null) {
```

Comment:
Actually, since this is executed in the context of the event loop so throwing an Http2Exception is more useful since it will properly shutdown the connection.